addons/stock/models/stock_quant.py

```diff
@@ -555,13 +555,15 @@ def _onchange_serial_number(self):
     def _onchange_product_id(self):
         if self.location_id:
             return
+        company_id = self.company_id and self.company_id.id or self.env.company.id
```

Comment:
The quant search will already filter based on selected companies unless we specifically only want to auto-select the location based on the main company selected (i.e. the one showing in the control panel). I would think we would want it to check the quants of all selected companies (if multi-company), unless you think otherwise.

Reply:
Yes you are right, I'll remove the part with the `company_id`.